crates/flipperzero/src/furi/stream_buffer.rs

```diff
@@ -0,0 +1,439 @@
+        self.buffer_ref.spaces_available()
+    }
+
+    /// Check if the underlying stream buffer is full.
+    pub fn is_full(&self) -> bool {
+        self.buffer_ref.is_full()
+    }
+
+    /// Check if the underlying stream buffer is empty.
+    pub fn is_empty(&self) -> bool {
+        self.buffer_ref.is_empty()
+    }
+
+    /// Attempt to reset the underlying stream buffer.
+    ///
+    /// This will clear the buffer, discarding any data it contains and returning it to its
+    /// initial empty state.
+    /// The reset can only occur if there are no tasks blocked waiting to send to or receive from
+    /// the stream buffer; attempting to reset during this time will result in an [`Err`].
+    pub fn reset(&mut self) -> EmptyResult {
+        self.buffer_ref.reset()
+    }
+}
+
+/// Pure additional methods to work with the [`Sender`] and `Receiver`.
+impl Receiver {
+    /// Check if the associated sender is still alive.
+    ///
+    /// This method helps prevent unnecessary data reception when the sender is no longer available.
+    pub fn is_sender_alive(&self) -> bool {
+        // SAFETY:
+        // Since both Receiver and Sender are not Clone, the only Arcs are those created by the
+        // stream_buffer function.
+        // If the strong count of the Arc referencing the buffer is 2, it indicates that
+        // the Receiver and the related Sender are still alive.
+        Arc::strong_count(&self.buffer_ref) == 2
+    }
+}
+
+/// Create a [`Sender`]/[`Receiver`] pair for a furi stream buffer.
+///
+/// This function initializes a furi stream buffer and provides access through the returned
+/// [`Sender`] and [`Receiver`] instances.
+///
+/// The `size` parameter must be a non-zero value, as the stream buffer requires a valid size
+/// to allocate memory.
+/// The `furi_stream_buffer_alloc` function enforces this constraint by rejecting zero sizes.
+///
+/// The `trigger_level` defines the number of bytes that must be present in the stream buffer
+/// before any blocked tasks waiting for data can proceed.
+///
+/// The Furi stream buffer is designed for single-task or single-interrupt access for both
+/// writing and reading operations.
+/// To enforce this, it splits the buffer into two parts, similar to the `mpsc::Sender` and
+/// `mpsc::Receiver` from the standard library.
+/// Both the sender and receiver maintain an owned reference to the stream buffer, ensuring it
+/// remains alive as long as either one exists.
+/// They can verify if their corresponding pair is still alive using [`Sender::is_receiver_alive`]
+/// and [`Receiver::is_sender_alive`] respectively.
+///
+/// Additionally, both the sender and receiver implement [`Send`] to allow transfer between
+/// threads.
+/// However, they explicitly do not implement [`Sync`] or [`Clone`] to guarantee that only one
+/// writer and one reader can exist at any given time per task.
+pub fn stream_buffer(size: NonZeroUsize, trigger_level: usize) -> (Sender, Receiver) {
```

Comment:
Perhaps `stream` might be a better name, since you're not actually getting a `StreamBuffer`—you're getting an abstraction on top of it.

Reply:
This is replaced with `StreamBuffer::into_stream` now.